newsrevenuehub/rev-engine
# redo DEV-1577 SPA - refactor code to remove paymentProvider redundant code

See https://github.com/newsrevenuehub/rev-engine/pull/501

Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)
fixes the white screen that shows up if paymentprovider is not connected

#### What's this PR do?
does the following
1. fixes the white screen
2. adds a logout link to the generic error page

#### Why are we doing this? How does it help us?
better user experience

#### How should this be manually tested?
To test
1. Login as various roles and the dashboard should show up
2. Remove a payment-provider for an RP, this should not lead to an error/ white screen showing up
3. A logout link will show in generic error page, this is not easily testable as its unlikely to happen normally. So a cypress test covers this

This PR also refactors/removes code

#### Have automated unit tests been added?
no

#### How should this change be communicated to end users?
na

#### Are there any smells or added technical debt to note?
no

#### Has this been documented? If so, where?
no

#### What are the relevant tickets? Add a link to any relevant ones.
https://news-revenue-hub.atlassian.net/browse/DEV-1577

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

#### Are there next steps? If so, what? Have tickets been opened for them?
https://news-revenue-hub.atlassian.net/browse/DEV-2042
This would handle the flows around publishing based on
